multiversx_sdk_cli/delegation/staking_provider.py

```diff
@@ -10,440 +9,847 @@
+        tx.version = version
+        tx.options = options
+        tx.guardian = guardian_address
+        tx.relayer = relayer_address
+
+        if gas_limit:
+            tx.gas_limit = gas_limit
+
+        self.sign_transaction(
+            transaction=tx,
+            sender=owner,
+            guardian=guardian_account,
+            relayer=relayer_account,
+            guardian_service_url=guardian_service_url,
+            guardian_2fa_code=guardian_2fa_code,
         )
-        tx.nonce = int(args.nonce)
-        tx.version = int(args.version)
-        tx.options = int(args.options)
-        tx.guardian = args.guardian
-
-        if args.gas_limit:
-            tx.gas_limit = int(args.gas_limit)
-
-        tx.signature = owner.sign_transaction(tx)
 
         return tx
 
-    def prepare_transaction_for_modifying_delegation_cap(self, owner: IAccount, args: Any) -> Transaction:
-        delegation_contract = Address.new_from_bech32(args.delegation_contract)
-
+    def prepare_transaction_for_modifying_delegation_cap(
+        self,
+        owner: IAccount,
+        delegation_contract: Address,
+        delegation_cap: int,
+        gas_limit: int,
+        gas_price: int,
+        value: int,
+        nonce: int,
+        version: int,
+        options: int,
+        guardian_account: Optional[IAccount] = None,
+        guardian_address: Optional[Address] = None,
+        relayer_account: Optional[IAccount] = None,
+        relayer_address: Optional[Address] = None,
+        guardian_service_url: str = "",
+        guardian_2fa_code: str = "",
+    ) -> Transaction:
         tx = self._factory.create_transaction_for_modifying_delegation_cap(
             sender=owner.address,
             delegation_contract=delegation_contract,
-            delegation_cap=int(args.delegation_cap),
+            delegation_cap=delegation_cap,
+        )
+        tx.value = value
+        tx.gas_price = gas_price
+        tx.nonce = nonce
+        tx.version = version
+        tx.options = options
+        tx.guardian = guardian_address
+        tx.relayer = relayer_address
+
+        if gas_limit:
+            tx.gas_limit = gas_limit
+
+        self.sign_transaction(
+            transaction=tx,
+            sender=owner,
+            guardian=guardian_account,
+            relayer=relayer_account,
+            guardian_service_url=guardian_service_url,
+            guardian_2fa_code=guardian_2fa_code,
         )
-        tx.nonce = int(args.nonce)
-        tx.version = int(args.version)
-        tx.options = int(args.options)
-        tx.guardian = args.guardian
-
-        if args.gas_limit:
-            tx.gas_limit = int(args.gas_limit)
-
-        tx.signature = owner.sign_transaction(tx)
 
         return tx
 
-    def prepare_transaction_for_automatic_activation(self, owner: IAccount, args: Any) -> Transaction:
-        delegation_contract = Address.new_from_bech32(args.delegation_contract)
+    def prepare_transaction_for_automatic_activation(
```

Comment:
Heads-up: in Barnard, this might change (we'll see at that point): 
 - https://github.com/multiversx/mx-chain-go/pull/6793

Reply:
Indeed.